keepercommander/commands/aram.py

```diff
@@ -1584,78 +1584,75 @@ def clean_up():
-            rq = {
-                'command':      'get_audit_event_reports',
-                'scope':        'enterprise',
-                'report_type':  'span',
-                'event_type':   event_type,
-                'columns':      ['record_uid'],
-                'aggregate':    ['last_created'],
-                'limit':        API_EVENT_SUMMARY_ROW_LIMIT
-            }
-            ts_lookup = dict()
-            while record_uids:
-                chunk = record_uids[:API_EVENT_SUMMARY_ROW_LIMIT]
-                record_uids = record_uids[API_EVENT_SUMMARY_ROW_LIMIT:]
-                rq_filter = {'record_uid': chunk, 'created': {'min': ts_floor}}
-                rq['filter'] = rq_filter
-                rs = api.communicate(params, rq)
-                events = rs['audit_event_overview_report_rows']
-                chunk_ts_lookup = {event['record_uid']: int(event['last_created']) for event in events}
-                ts_lookup.update(chunk_ts_lookup)
-            return ts_lookup
-
-        def update_record_aging(params, sd, r_uids, event_type, set_ent_field_fn, on_updated, msg, ts_floor=0):
-            # type: (KeeperParams, sox_data.SoxData, List[int], str, Callable, Callable, str, int) -> None
-            logging.info(msg)
-            ts_lookup = get_record_event_dates(params, r_uids, event_type, ts_floor)
-            updated_entities = []
-            for rec_uid in ts_lookup:
-                entity = sd.storage.record_aging.get_entity(rec_uid) or StorageRecordAging(rec_uid)
-                entity = set_ent_field_fn(entity, ts_lookup.get(rec_uid))
-                updated_entities.append(entity)
+    @staticmethod
+    def update_aging_data(params, sox, period_start_ts):   # type: (KeeperParams, sox_data.SoxData, int) -> None
+        from_date = min(sox.storage.records_dated, sox.storage.last_pw_audit)
+        if from_date > period_start_ts:
+            return
+        if from_date == 0:
+            min_dt = datetime.datetime.now() - datetime.timedelta(days=365) * 5
+            from_date = int(min_dt.timestamp())
+
+        audit_filter = {
+            'audit_event_type': ['record_add', 'record_password_change'],
+            'created': {'min': 0}
+        }
+        rq = {
+            'command':      'get_audit_event_reports',
+            'scope':        'enterprise',
+            'report_type':  'span',
+            'columns':      ['record_uid', 'audit_event_type'],
+            'aggregate':    ['last_created'],
+            'filter':       audit_filter,
+            'order':        'ascending',
+            'limit':        API_EVENT_SUMMARY_ROW_LIMIT
+        }
+        changed_records = {}    # type: Dict[str, StorageRecordAging]
+        last_added = 0
+        last_password_change = 0
+        done = False
+        logging.info('Loading record password change information...')
+        while not done:
+            audit_filter['created']['min'] = from_date
```

Comment:
the upper limit (and not the lower limit) should be adjusted upon each subsequent request (instead of setting the "min" of the "created" property, we should be setting the "max" to the least recent timestamp of the last set of returned event rows)